tasks/migrated-component-scanner.js

```diff
@@ -0,0 +1,199 @@
+	try {
+		const componentsDir = path.resolve(process.cwd(), "components");
+		const directories = getAllComponentDirectories();
+
+		if (!statusType) return directories;
+
+		// Filter directories that have status type in their stories
+		const matchingComponents = directories.filter(dir => {
+			const storiesDir = path.join(componentsDir, dir, "stories");
+
+			// Check if stories directory exists
+			if (!fs.existsSync(storiesDir)) return false;
+
+			// Get all story files
+			const storyFiles = fs.readdirSync(storiesDir)
+				.filter(file => file.endsWith(".stories.js"));
+
+			// Check each story file for status type
+			return storyFiles.some(file => {
+				const storyContent = fs.readFileSync(path.join(storiesDir, file), "utf8");
+				return storyContent.includes(`type: "${statusType}"`);
+			});
+		});
+
+		return matchingComponents;
+	} catch (error) {
+		console.error(`Error getting components with status ${statusType}:`, error);
+		return [];
+	}
+}
+
+/**
+ * Extracts the title from a Storybook .stories.js file (to display the title in a list with proper capitalization)
+ * @param {string} storyFilePath - Absolute path to the .stories.js file
+ * @returns {string|null} The extracted title, or null if not found
+ */
+function extractTitleFromStoryFile(storyFilePath) {
+	try {
+		const content = fs.readFileSync(storyFilePath, "utf8");
+		const match = content.match(/title:\s*["'`](.+?)["'`]/);
+		if (match && match[1]) {
+			return match[1];
+		}
+	} catch (error) {
+		console.warn(`Could not extract title from ${storyFilePath}:`, error);
+	}
+	return null;
+}
+
+/**
+ * Generates the URL fragment from a component's title
+ * @param {string} title - The title to generate a URL fragment from
+ * @returns {string} The generated URL fragment
+ */
+function generateUrlFragmentFromTitle(title) {
+	return title
+		.split("/")
+		.map(segment =>
+			segment
+				.trim()
+				.toLowerCase()
+				.replace(/[^a-z0-9 -]/g, "") // remove special chars except space and dash (i.e "in-line alert")
+				.replace(/\s+/g, "-")       // replace spaces with dash
+		)
+		.join("-")
+		.replace(/-+/g, "-"); // collapse multiple dashes
+}
+
+/**
+ * Generates a list of migrated components with their titles, and docs links
+ * @returns {Object} Information about migrated components
+ */
+function generateMigratedComponentsReport() {
+	const allComponents = getAllComponentDirectories();
+	const migratedComponents = getComponentsByStatus("migrated");
+
+	const componentsWithData = migratedComponents.map((dir) => {
+		const storiesDir = path.resolve(process.cwd(), "components", dir, "stories");
+		if (!fs.existsSync(storiesDir)) {
+			return {
+				name: dir,
+				title: dir,
+				url: dir,
+			};
+		}
+
+		const storyFiles = fs.readdirSync(storiesDir).filter(file => file.endsWith(".stories.js"));
+		let title = null;
+		for (const file of storyFiles) {
+			title = extractTitleFromStoryFile(path.join(storiesDir, file));
+			if (title) break;
```

Comment:
I think the logic around here needs updating if we want to be able to pick up both text field and text area. When I log `storyFiles` for text field I get `[ 'textarea.stories.js', 'textfield.stories.js' ]`, but then we have it set up to only extract one title and in this weird edge case we have two.

I think that's the only place where we have that situation though. I wonder if there are other places where this is potentially causing an issue? If so we could consider modifying the way textfield and textarea are packaged?

My vote based on what I know currently would be trying to update the logic here rather than doing that.

Reply:
Great call, thank you so much. Yes, I think it's the only place right now. We separated meter from progress bar, and those would have been in the same boat. 

Anyways. the script has been updated to account for multiple story files in a directory. 👍 

<img width="94" height="63" alt="Screenshot 2025-08-15 at 5 18 06 PM" src="https://github.com/user-attachments/assets/c538fb31-cc8b-4f13-8bdd-cff1d973f377" />